core/api/src/mill/api/internal/README.md

```diff
@@ -0,0 +1,4 @@
+These `*ModuleApi` traits are only allowed to use Scala standard library to decouple the `build.mill` code from
+the `mill.runner` code.
+
+https://github.com/com-lihaoyi/mill/pull/4879
```

Comment:
This duplicates docs already present in the `core/api/package.mill` scaladoc. Can we combine them there? The build scaladoc is visible to `./mill inspect` and in the IDE when working with build files, which is better than the readme.md which doesn't integrate with anything